pkgs/by-name/so/sonarr/package.nix

```diff
@@ -0,0 +1,163 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, substituteAll
+, buildDotnetModule
+, dotnetCorePackages
+, sqlite
+, withFFmpeg ? true # replace bundled ffprobe binary with symlink to ffmpeg package.
+, ffmpeg
+, fetchYarnDeps
+, yarn
+, nodejs
+, nixosTests
+  # update script
+, writeShellScript
+, writers
+, python3Packages
+, nix
+, prefetch-yarn-deps
+}:
+let
+  ffprobe = lib.getExe' ffmpeg "ffprobe";
+in
+buildDotnetModule rec {
+  pname = "sonarr";
+  version = "4.0.2.1183";
+
+  src = fetchFromGitHub {
+    owner = "Sonarr";
+    repo = "Sonarr";
+    rev = "v${version}";
+    hash = "sha256-EJ2XDlcVdUH0IHNlXjVm02wrmpjJVK3WKVifxZ63Lxk=";
+  };
+
+  patches = [
+    (substituteAll {
+      src = ./buildprops.patch;
+      copyrightYear = "2024";
+      assemblyVersion = version;
+      assemblyConfiguration = "main";
+    })
+  ];
+
+  strictDeps = true;
+  nativeBuildInputs = [ nodejs yarn prefetch-yarn-deps ];
+
+  yarnOfflineCache = fetchYarnDeps {
+    yarnLock = src + "/yarn.lock";
+    hash = "sha256-N/yeKkYTpOTkwl2o9YjfioeBxUf3YSZ692nuQcb7PzM=";
+  };
+
+  postConfigure = ''
+    yarn config --offline set yarn-offline-mirror $yarnOfflineCache
+    fixup-yarn-lock yarn.lock
+    yarn install --offline --frozen-lockfile --ignore-platform --ignore-scripts --no-progress --non-interactive
+    patchShebangs node_modules
+  '';
+  postBuild = ''
+    yarn --offline run build --env production
+  '';
+  postInstall = lib.optionalString (withFFmpeg) ''
+    rm $out/lib/sonarr/ffprobe
+    ln -s ${lib.escapeShellArg ffprobe} $out/lib/sonarr/ffprobe
+  '' + ''
+    cp -a _output/UI $out/lib/sonarr/UI
+  '';
+  # Add an alias for compatibility with Sonarr v3 package.
+  postFixup = ''
+    ln -s Sonarr $out/bin/NzbDrone
+  '';
+
+  nugetDeps = ./deps.nix;
+  nugetConfigFile = "src/NuGet.Config";
```

Comment:
I think instead of adding support for `nugetConfigFile`, it could be a lot simpler to either:

- fix whatever is not finding the config file in src/

or 

- copy/link src/NuGet.Config into the root 

My worry is that any use of dotnet tools that isn't aware of this flag will be broken.

Reply:
I agree with this, moving the file to the root would be much better than changing the entire dotnet infra to accomodate a single pacakge.

Reply:
We need this for `nuget-to-nix` that uses `dotnet nuget list source` and therefore does not search for NuGet configuration along the project file path. Perhaps we can teach `nuget-to-nix` to do that?

Reply:
Added a patch to move NuGet.config to the source root.